pkg/lib/util/ssa.go

```diff
@@ -0,0 +1,187 @@
+package util
```

Comment:
I do think this needs to go in a different package. `lib/util` is a little to generic, especially when we _could_ have `lib/controller-runtime/client` package or something similarly descriptive.